### PR TITLE
block_synchronizer/deploy_acquisition: add unit tests

### PR DESCRIPTION
**Pull request by alsrdn**
Unit tests for the deploy acquisition logic as part of https://github.com/casper-network/casper-node/issues/3437.

### CODE REVIEW

File: node/src/components/block_synchronizer/deploy_acquisition/tests.rs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            .map(|(deploy_hash, deploy)| {
+                DeployId::new(*deploy_hash, deploy.approvals_hash().unwrap())
+            })
+            .collect(),
+        false,
+    ));
+
+    assert_matches!(
+        deploy_acquisition.apply_approvals_hashes(&approvals_hashes),
+        Err(Error::AcquisitionByIdNotPossible)
```
Comment (rafal-ch) [inline node/src/components/block_synchronizer/deploy_acquisition/tests.rs:58]: Can you add a test covering the [EncounteredNonVacantDeployState](https://github.com/casper-network/casper-node/blob/dev/node/src/components/block_synchronizer/deploy_acquisition.rs#L15) error variant?

**Review by rafal-ch — APPROVED** [review]
LGTM :+1: 
with one minor suggestion :)

**Review by Fraser999 — APPROVED** [review]
Approved assuming Rafal's ask is satisfied.

File: node/src/components/block_synchronizer/deploy_acquisition/tests.rs (hunk 1 above)
Comment (alsrdn) [reply node/src/components/block_synchronizer/deploy_acquisition/tests.rs:58]: added here: https://github.com/casper-network/casper-node/pull/3690/commits/c5117c8e0139028ad0e64f020d7e8e06b04143cc

**Comment by alsrdn** [comment]
bors try

**Pull request merged**